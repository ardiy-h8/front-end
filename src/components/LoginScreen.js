import React, { Component } from 'react'
import { loginWithGoogle, loginWithFacebook } from '../helpers/auth'
import { firebaseAuth } from '../config/constants'
import { Button, Grid, Paper } from 'material-ui'

const firebaseAuthKey = 'firebaseAuthInProgress'
const appTokenKey = 'appToken'

class LoginScreen extends Component {
  constructor (props) {
    super(props)
    this.state = {
      splashScreen: false
    }
    // this.handleGooleLogin = this.handleGooleLogin.bind(this)
  }
  handleGoogleLogin () {
    loginWithGoogle().catch(err => {
      alert(err)
      localStorage.removeItem(firebaseAuth)
    })
    localStorage.setItem(firebaseAuthKey, '1')
  }
  handleFacebookLogin () {
    loginWithFacebook().catch(err => {
      alert(err)
      localStorage.removeItem(firebaseAuth)
      localStorage.setItem(firebaseAuthKey, '1')
    })
  }
  componentWillMount () {
    if (localStorage.getItem(appTokenKey)) {
      this.props.history.push('/home')
      return
    }
    firebaseAuth().onAuthStateChanged(user => {
      if (user) {
        console.log('User signed in: ', JSON.stringify(user))
        localStorage.removeItem(firebaseAuthKey)
        localStorage.setItem(appTokenKey, user.uid)
        this.props.history.push('/home')
      }
    })
  }
  render () {
    return (
      <div style={styles.root}>
        <Grid container spacing={24}>
          <Grid item xs={12}>
            <div style={styles.bg}></div>
            <Paper style={styles.paper}>
              <div>
                <img src="./assets/logo.png" alt='logo' style={styles.logo}/>
              </div>
              <div>
                <Button
                  style={styles.btnGoogle}
                  variant='raised'
                  onClick={() => this.handleGoogleLogin()}
                >
                  <i className='fab fa-google' style={{ marginRight: 30 }} />
                  Login With Google
                </Button>
              </div>
              <div>
                <Button
                  style={styles.btnFacebook}
                  variant='raised'
                  onClick={() => this.handleFacebookLogin()}
                >
                  <i
                    className='fab fa-facebook-f'
                    style={{ marginRight: 30 }}
                  />
                  Login With Facebook
                </Button>
              </div>
            </Paper>
          </Grid>
        </Grid>
      </div>
    )
  }
}
const styles = {
  root: {
    display: 'flex',
  },
  paper: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    height: '100vh',
    zIndex: 9999
  },
  bg: {
    position: 'fixed',
    background: `url('./assets/login-bg.jpg') no-repeat center center fixed`,
    backgroundSize: 'cover',
    height: '100%',
    width: '100%',
    filter: 'blur(5px)',
  },
  btnGoogle: {
    marginTop: 20,
    width: 300,
    backgroundColor: '#EA4335',
    textAlign: 'left',
    color: 'white'
  },
  btnFacebook: {
    marginTop: 20,
    width: 300,
    backgroundColor: '#3b5998',
    color: 'white'
  },
  logo: {
    width: 180,
    position: 'relative'
  }
}

export default LoginScreen
