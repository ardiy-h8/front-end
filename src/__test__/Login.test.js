import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount, shallow, configure } from 'enzyme';


import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });


import LoginScreen from '../components/LoginScreen'
import { Button, Grid, Paper } from 'material-ui'



var login =  new LoginScreen()


describe('google login button', () => {
  
  it('Should be defined', () => {
    expect(LoginScreen).toBeDefined()
  })

  it('should render correctly', () => {
    const tree = shallow(
      <Button
        variant='raised'
        onClick={() => this.handleGoogleLogin()}
      >
        <i className='fab fa-google'/>
        Login With Google
        </Button>
    )
    expect(tree).toMatchSnapshot()
    
  })

  it('should render correctly', () => {
    const tree = shallow(
      <Button
        variant='raised'
        onClick={() => this.handleFacebookLogin()}
      >
        <i className='fab fa-google' />
        Login With Facebook
        </Button>
    )
    expect(tree).toMatchSnapshot()
  })

  // simulate click button
  it('should call mock function when button is clicked', () => {
    const tree = shallow(
      <Button
        variant='raised'
        onClick={() => login.handleGoogleLogin()}
      >
        <i className='fab fa-google' />
        Login With Google
        </Button>
    )
    tree.simulate('click')
  })
  
  // it('should call mock function when button is clicked', () => {
  //   const mockFn = login.handleFacebookLogin
  //   const tree = shallow(
  //     <Button
  //       variant='raised'
  //       onClick={mockFn}
  //     >
  //     <i
  //       className='fab fa-facebook-f'
  //     />
  //     Login With Facebook
  //     </Button>
  //   )
  //   tree.simulate('click')

  // })

})

class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
};

global.localStorage = new LocalStorageMock;