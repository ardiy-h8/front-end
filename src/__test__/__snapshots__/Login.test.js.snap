// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`google login button should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Button)
    onClick={[Function]}
    variant="raised"
>
    <i
        className="fab fa-google"
    />
    Login With Google
</WithStyles(Button)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <i
          className="fab fa-google"
/>,
        "Login With Google",
      ],
      "classes": Object {
        "colorInherit": "MuiButton-colorInherit-5",
        "disabled": "MuiButton-disabled-10",
        "fab": "MuiButton-fab-11",
        "flatPrimary": "MuiButton-flatPrimary-3",
        "flatSecondary": "MuiButton-flatSecondary-4",
        "fullWidth": "MuiButton-fullWidth-15",
        "keyboardFocused": "MuiButton-keyboardFocused-7",
        "label": "MuiButton-label-2",
        "mini": "MuiButton-mini-12",
        "raised": "MuiButton-raised-6",
        "raisedPrimary": "MuiButton-raisedPrimary-8",
        "raisedSecondary": "MuiButton-raisedSecondary-9",
        "root": "MuiButton-root-1",
        "sizeLarge": "MuiButton-sizeLarge-14",
        "sizeSmall": "MuiButton-sizeSmall-13",
      },
      "color": "default",
      "disableFocusRipple": false,
      "disableRipple": false,
      "disabled": false,
      "fullWidth": false,
      "mini": false,
      "onClick": [Function],
      "size": "medium",
      "type": "button",
      "variant": "raised",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fab fa-google",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      "Login With Google",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <i
            className="fab fa-google"
/>,
          "Login With Google",
        ],
        "classes": Object {
          "colorInherit": "MuiButton-colorInherit-5",
          "disabled": "MuiButton-disabled-10",
          "fab": "MuiButton-fab-11",
          "flatPrimary": "MuiButton-flatPrimary-3",
          "flatSecondary": "MuiButton-flatSecondary-4",
          "fullWidth": "MuiButton-fullWidth-15",
          "keyboardFocused": "MuiButton-keyboardFocused-7",
          "label": "MuiButton-label-2",
          "mini": "MuiButton-mini-12",
          "raised": "MuiButton-raised-6",
          "raisedPrimary": "MuiButton-raisedPrimary-8",
          "raisedSecondary": "MuiButton-raisedSecondary-9",
          "root": "MuiButton-root-1",
          "sizeLarge": "MuiButton-sizeLarge-14",
          "sizeSmall": "MuiButton-sizeSmall-13",
        },
        "color": "default",
        "disableFocusRipple": false,
        "disableRipple": false,
        "disabled": false,
        "fullWidth": false,
        "mini": false,
        "onClick": [Function],
        "size": "medium",
        "type": "button",
        "variant": "raised",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fab fa-google",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "Login With Google",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`google login button should render correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Button)
    onClick={[Function]}
    variant="raised"
>
    <i
        className="fab fa-google"
    />
    Login With Facebook
</WithStyles(Button)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <i
          className="fab fa-google"
/>,
        "Login With Facebook",
      ],
      "classes": Object {
        "colorInherit": "MuiButton-colorInherit-5",
        "disabled": "MuiButton-disabled-10",
        "fab": "MuiButton-fab-11",
        "flatPrimary": "MuiButton-flatPrimary-3",
        "flatSecondary": "MuiButton-flatSecondary-4",
        "fullWidth": "MuiButton-fullWidth-15",
        "keyboardFocused": "MuiButton-keyboardFocused-7",
        "label": "MuiButton-label-2",
        "mini": "MuiButton-mini-12",
        "raised": "MuiButton-raised-6",
        "raisedPrimary": "MuiButton-raisedPrimary-8",
        "raisedSecondary": "MuiButton-raisedSecondary-9",
        "root": "MuiButton-root-1",
        "sizeLarge": "MuiButton-sizeLarge-14",
        "sizeSmall": "MuiButton-sizeSmall-13",
      },
      "color": "default",
      "disableFocusRipple": false,
      "disableRipple": false,
      "disabled": false,
      "fullWidth": false,
      "mini": false,
      "onClick": [Function],
      "size": "medium",
      "type": "button",
      "variant": "raised",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fab fa-google",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      "Login With Facebook",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <i
            className="fab fa-google"
/>,
          "Login With Facebook",
        ],
        "classes": Object {
          "colorInherit": "MuiButton-colorInherit-5",
          "disabled": "MuiButton-disabled-10",
          "fab": "MuiButton-fab-11",
          "flatPrimary": "MuiButton-flatPrimary-3",
          "flatSecondary": "MuiButton-flatSecondary-4",
          "fullWidth": "MuiButton-fullWidth-15",
          "keyboardFocused": "MuiButton-keyboardFocused-7",
          "label": "MuiButton-label-2",
          "mini": "MuiButton-mini-12",
          "raised": "MuiButton-raised-6",
          "raisedPrimary": "MuiButton-raisedPrimary-8",
          "raisedSecondary": "MuiButton-raisedSecondary-9",
          "root": "MuiButton-root-1",
          "sizeLarge": "MuiButton-sizeLarge-14",
          "sizeSmall": "MuiButton-sizeSmall-13",
        },
        "color": "default",
        "disableFocusRipple": false,
        "disableRipple": false,
        "disabled": false,
        "fullWidth": false,
        "mini": false,
        "onClick": [Function],
        "size": "medium",
        "type": "button",
        "variant": "raised",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fab fa-google",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "Login With Facebook",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
